File: frontend/src/types/video.ts
/**
 * 视频相关数据类型定义
 */

// 视频标签类型
export interface VideoTag {
  name: string;
  query: string;
}

// 视频流地址类型
export interface StreamUrl {
  quality: string;
  url: string;
}

// 视频制作商/发行商
export interface VideoStudio {
  name: string;
  icon_url?: string;
  url?: string;
  query?: string;
}

// 视频类型
export interface VideoType {
  name: string;
  query?: string;
}

// 基础视频信息
export interface VideoBase {
  video_id: string;
  title: string;
  cover_url: string;
}

// Banner视频
export interface BannerVideo extends VideoBase {
  description?: string;
}

// 视频预览信息（详细视频）
export interface VideoPreview extends VideoBase {
  duration?: string;
  view_count?: number;
  like_rate?: string;
  like_count?: number;
  studio?: VideoStudio;
}

// 视频详情
export interface VideoDetail extends VideoPreview {
  description?: string;
  subtitle?: string;
  video_type?: VideoType;
  default_video_url?: string;
  stream_urls: StreamUrl[];
  tags: VideoTag[];
  upload_date?: string;
  views?: string;
  series_videos?: VideoPreview[];
  basic_related_videos?: VideoBase[];
  detailed_related_videos?: VideoPreview[];
}

// 视频分类数据
export interface VideoSection {
  title: string;
  search_suffix: string;
  videos: VideoBase[] | VideoPreview[];
}

// 首页数据响应
export interface HomeData {
  banners: BannerVideo;
  latest_videos: VideoSection[];
  new_arrivals_videos: VideoSection[];
  new_uploads_videos: VideoSection[];
  chinese_subtitle_videos: VideoSection[];
  popular_videos: VideoSection[];
  daily_rank_videos: VideoSection[];
  monthly_rank_videos: VideoSection[];
  error?: string;
}

// 视频评论
export interface VideoComment {
  comment_id?: string;
  user_avatar?: string;
  username: string;
  comment_time?: string;
  comment_content: string;
  like_count?: number;
  reply_count?: number;
}

// 评论回复
export interface CommentReply {
  user_avatar?: string;
  username: string;
  reply_time?: string;
  reply_content: string;
  like_count?: number;
}

// 搜索结果
export interface SearchResults {
  total_pages: number;
  page: number;
  has_next: boolean;
  basic_videos: VideoBase[];
  detailed_videos: VideoPreview[];
}

// 搜索组合选项
export interface SearchCombination {
  video_types: string[];
  tags: Record<string, string[]>;
  sort: string[];
} 

File: backend/app/models/video.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime


class VideoTag(BaseModel):
    """标签模型"""
    name: str
    query: str = ""


class VideoStreamUrl(BaseModel):
    """视频流URL模型"""
    quality: str
    url: str


class VideoStudio(BaseModel):
    """视频制作商/发行商模型"""
    name: str
    icon_url: Optional[str] = ""
    url: Optional[str] = ""
    query: Optional[str] = ""


class VideoType(BaseModel):
    """视频类型模型"""
    name: str
    query: Optional[str] = ""


class VideoBase(BaseModel):
    """视频基础模型"""
    video_id: str
    title: str
    cover_url: Optional[str] = ""


# 首页Banner视频模型
class BannerVideo(VideoBase):
    description: Optional[str] = ""


class VideoPreview(VideoBase):
    """视频预览模型，用于列表展示"""
    duration: Optional[str] = ""
    view_count: Optional[int] = 0
    like_rate: Optional[str] = ""
    like_count: Optional[int] = 0
    studio: Optional[VideoStudio] = None


class VideoDetail(VideoPreview):
    """视频详情模型"""
    description: Optional[str] = ""
    subtitle: Optional[str] = ""
    upload_date: Optional[datetime] = ""
    video_type: Optional[VideoType] = None
    default_video_url: Optional[str] = ""
    stream_urls: List[VideoStreamUrl] = []
    tags: List[VideoTag] = []
    series_videos: Optional[List[VideoPreview]] = []
    # 基础相关视频（仅包含基本信息）
    basic_related_videos: List[VideoBase] = Field(default_factory=list, description="基础相关视频推荐，仅包含基本信息")
    # 详细相关视频（包含更多详细信息）
    detailed_related_videos: List[VideoPreview] = Field(default_factory=list, description="详细相关视频推荐，包含更多信息如时长、点赞等")


class HomeData(BaseModel):
    """首页数据模型"""
    banners: BannerVideo = None
    latest_videos: List[Dict[str, Any]] = []
    new_arrivals_videos: List[Dict[str, Any]] = []
    new_uploads_videos: List[Dict[str, Any]] = []
    popular_videos: List[Dict[str, Any]] = []
    ai_generated_videos: List[Dict[str, Any]] = []
    bubble_tea_videos: List[Dict[str, Any]] = []
    error: Optional[str] = None


class VideoComment(BaseModel):
    """视频评论模型"""
    comment_id: Optional[str] = ""
    user_avatar: Optional[str] = ""
    username: str
    comment_time: Optional[str] = ""
    comment_content: str
    like_count: Optional[int] = 0
    reply_count: Optional[int] = 0


class CommentReply(BaseModel):
    """评论回复模型"""
    user_avatar: Optional[str] = ""
    username: str
    reply_time: Optional[str] = ""
    reply_content: str
    like_count: Optional[int] = 0


class SearchCombination(BaseModel):
    """搜索组合模型"""
    video_types: List[str] = Field(default_factory=list, description="视频类型选项")
    tags: Dict[str, List[str]] = Field(default_factory=dict, description="按分类组织的标签选项")
    sort: List[str] = Field(default_factory=list, description="排序方式选项")


class SearchResults(BaseModel):
    """搜索结果模型"""
    total_pages: int = 0
    page: int = 1
    basic_videos: List[VideoBase] = []
    detailed_videos: List[VideoPreview] = []
    has_next: bool = False
